operator/internal/controller/konfluxreleaseservice_controller.go

```diff
@@ -126,6 +140,35 @@ func (r *KonfluxReleaseServiceReconciler) applyManifests(ctx context.Context, ow
 	return nil
 }
 
+// applyReleaseServiceDeploymentCustomizations applies user-defined customizations to ReleaseService deployments.
+func applyReleaseServiceDeploymentCustomizations(deployment *appsv1.Deployment, spec konfluxv1alpha1.KonfluxReleaseServiceSpec) error {
+	switch deployment.Name {
+	case releaseControllerManagerDeploymentName:
+		if spec.ReleaseControllerManager != nil {
+			deployment.Spec.Replicas = &spec.ReleaseControllerManager.Replicas
+		}
+		if err := buildReleaseControllerManagerOverlay(spec.ReleaseControllerManager).ApplyToDeployment(deployment); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// buildReleaseControllerManagerOverlay builds the pod overlay for the controller-manager deployment.
+func buildReleaseControllerManagerOverlay(spec *konfluxv1alpha1.ControllerManagerDeploymentSpec) *customization.PodOverlay {
+	if spec == nil {
+		return customization.NewPodOverlay()
+	}
+
+	return customization.BuildPodOverlay(
+		customization.DeploymentContext{Replicas: spec.Replicas},
+		customization.WithContainerBuilder(
+			releaseManagerContainerName,
```

Comment:
I think that I've missed it in the pr for integration service, but for most controllers when the number of replicas is >1, leader election should be enabled, thus you would need to add this option as well https://github.com/konflux-ci/konflux-ci/blob/main/operator/pkg/customization/container.go#L98